### PR TITLE
Setup.js

### PR DESCRIPTION
**Pull request by ghost**
- [x] - Added a try catch, so no need to run `setup.js` at first.

I have tested it on this version, and it works perfectly fine for me :)

### CODE REVIEW

**Comment by eslachance** [comment]
It's a nice thought except that setup.js doesn't restart the app, so after running it the config will not be available. Perhaps adding a process.exit() at the end of setup could fix it? 

Also, don't remove the line from the readme - it's useful to have it there, since someone running setup.js *before* opening the bot is still going to have a better experience.

File: README.md [hunk 1]
```diff
@@ -43,7 +43,6 @@ Once finished:
 
 - In the folder from where you ran the git command, run `cd guidebot` and then run `npm install`
 - **If you get any error about python or msibuild.exe or binding, read the requirements section again!**
-- Run `node setup.js` to generate a proper configuration file and settings.
```
Comment (eslachance) [inline README.md:46]: Don't remove this line, it's still useful and relevant.

File: index.js [hunk 2]
```diff
@@ -3,6 +3,14 @@
 // you.
 if (Number(process.version.slice(1).split(".")[0]) < 12) throw new Error("Node 12.0.0 or higher is required. Update Node on your system.");
 
+try {
+  require("./config.js");
+} catch {
+  console.log("Creating a config.js file for the bot");
+  require("./setup.js");
```
Comment (eslachance) [inline index.js:10]: After running the setup the app will have to restart, so within setup.js you can process.exit() in order to make sure the bot is restarted again.

**Review by eslachance — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ghost) [reply README.md:46]: Done in 7484f757d6ea4ec763942a6f399d74c76aa6be1d

File: index.js (hunk 2 above)
Comment (ghost) [reply index.js:10]: Resolved in 2716fe174311e9a54a1a8997ab1e247b94a8fa8a

**Comment by ghost** [comment]
There was a conflict, was resolved in b80413c75eb8562c8a3030970783677931b9488e. Be sure to check it out once before merging 👍 .

File: setup.js [hunk 3]
```diff
@@ -91,4 +91,5 @@ let prompts = [
   console.log("REMEMBER TO NEVER SHARE YOUR TOKEN WITH ANYONE!");
   console.log("Configuration has been written, enjoy!");
   await settings.close();
-}());
\ No newline at end of file
+}());
+process.exit();
```
Comment (eslachance) [inline setup.js]: Hold up. process.exit() outside the async function will make it exit before it even loads enmap. Please move this to line 94, and then it should be good!

**Review by eslachance — CHANGES_REQUESTED** [review]

File: setup.js (hunk 3 above)
Comment (ghost) [reply setup.js]: Oops, yes, my bad. Fixed it

**Review by eslachance — APPROVED** [review]

**Pull request merged**